dask/dataframe/core.py

```diff
@@ -3227,9 +3227,34 @@ def pivot_table(self, index=None, columns=None,
         return pivot_table(self, index=index, columns=columns, values=values,
                            aggfunc=aggfunc)
 
-    def to_records(self, index=False):
+    def to_records(self, lengths=None, index=False):
         from .io import to_records
-        return to_records(self)
+        from dask.array.core import normalize_chunks
+
+        if lengths is True:
```

Comment:
I've been regretting the use of `lengths=True`. I fear that it's not explicit enough.

What do people think about deprecating `lengths=True` and requiring something like `lengths='compute'` to make it clear what you're doing?

Reply:
I agree `lengths='compute'` is clearer. Will not add to this commit for now - may be worth creating a new issue to change all occurrences (e.g. with `.to_dask_array()`) + docs etc. ?